dsp-shared/src/main/scala/dsp/valueobjects/Role.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package dsp.valueobjects
+
+import dsp.errors.BadRequestException
+import zio.prelude.Validation
+
+object Role {
+
+  /**
+   * LangString value object.
+   *
+   * @param value the [[String]] the value to be validated
+   * @param isoCode thelanguage ISO code to be validated
+   */
+  sealed abstract case class LangString private (value: String, isoCode: String)
```

Comment:
I think it would be more intuititve to turn the paramters' order around: have the isoCode and then the value. This would be the same order as the value is represented when given by or returned to the user, e.g. {"en": "some value"}.

Reply:
Subject to discuss.